spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+require 'pry'
+
+
+describe CommentsController do
+
+  let(:user){create(:user)}
+
+  before :each do
+    cookies[:auth_token] = user.auth_token
+    user.posts = create_list(:post, 1)
+    @post= user.posts.first
+    allow(controller).to receive(:store_referer).
+                        and_return(session[:referer] = user_posts_path(user.id))
+  end
+
+  context 'logged in user' do
+
+    describe 'POST #create' do
+
+      it 'can create a comment with correct parameters' do
+
+        expect do
+          post :create, comment: attributes_for(:post_comment,
+                        user_id:          user.id,
+                        commentable_id:   @post.id,
+                        commentable_type: "Post")
+
+        end.to change(Comment, :count).by(1)
+
+      end
+
+      it 'redirects to index/refreshes after comment creation' do
+        post :create, comment: attributes_for(:post_comment,
+                        user_id:          user.id,
+                        commentable_id:   @post.id,
+                        commentable_type: "Post")
+        expect(response).to redirect_to (user_posts_path(user.id))
+      end
+
+      it "can create a comment on another user's timeline" do
+        another_user = create(:user)
+        expect do
+          post :create, comment: attributes_for(:post_comment,
+                        user_id:          user.id,
+                        commentable_id:   @post.id,
+                        commentable_type: "Post")
+
+        end.to change(Comment, :count).by(1)
+      end
+
+    end
+
+    describe 'DELETE #destroy' do
```

Comment:
I'm not 100% sure but I think that you can actually still set these up in `let` statements.

This way works just fine, though.